packages/mdc-menu/adapter.ts

```diff
@@ -21,95 +21,64 @@
  * THE SOFTWARE.
  */
 
-/* eslint no-unused-vars: [2, {"args": "none"}] */
+import {MenuItemEventDetail} from './types';
 
 /**
- * Adapter for MDC Menu. Provides an interface for managing
- * - selected element classes
- * - get focused elements
- * - toggling a checkbox inside a list item
- *
- * Additionally, provides type information for the adapter to the Closure
- * compiler.
- *
  * Implement this adapter for your framework of choice to delegate updates to
  * the component in your framework of choice. See architecture documentation
  * for more details.
  * https://github.com/material-components/material-components-web/blob/master/docs/code/architecture.md
- *
- * @record
  */
-class MDCMenuAdapter {
+interface MDCMenuAdapter {
   /**
    * Adds a class to the element at the index provided.
-   * @param {number} index
-   * @param {string} className
    */
-  addClassToElementAtIndex(index, className) {}
+  addClassToElementAtIndex(index: number, className: string): void;
 
   /**
    * Removes a class from the element at the index provided
-   * @param {number} index
-   * @param {string} className
    */
-  removeClassFromElementAtIndex(index, className) {}
+  removeClassFromElementAtIndex(index: number, className: string): void;
 
   /**
    * Adds an attribute, with value, to the element at the index provided.
-   * @param {number} index
-   * @param {string} attr
-   * @param {string} value
    */
-  addAttributeToElementAtIndex(index, attr, value) {}
+  addAttributeToElementAtIndex(index: number, attr: string, value: string): void;
 
   /**
    * Removes an attribute from an element at the index provided.
-   * @param {number} index
-   * @param {string} attr
    */
-  removeAttributeFromElementAtIndex(index, attr) {}
+  removeAttributeFromElementAtIndex(index: number, attr: string): void;
 
   /**
-   * Returns true if the element contains the className.
-   * @param {?HTMLElement} element
-   * @param {string} className
-   * @return {boolean} true if the element contains the className
+   * @return true if the element contains the className.
    */
-  elementContainsClass(element, className) {}
+  elementContainsClass(element: Element, className: string): boolean;
 
   /**
    * Closes the menu-surface.
    */
-  closeSurface() {}
+  closeSurface(): void;
 
   /**
-   * Returns the index for the element provided.
-   * @param {?HTMLElement} element
-   * @return {number} index of the element in the list or -1 if it is not in the list.
+   * @return Index of the element in the list or -1 if it is not in the list.
    */
-  getElementIndex(element) {}
+  getElementIndex(element: Element): number;
 
   /**
-   * Returns the parentElement of the provided element.
-   * @param {?HTMLElement} element
-   * @return {?HTMLElement} parentElement of the element provided.
+   * @return The parentElement of the provided element.
    */
-  getParentElement(element) {}
+  getParentElement(element: Element): Element | null;
```

Comment:
according to TS `Node` class it is an HTMLElement that is returned

Reply:
`lib.dom.d.ts` is wrong 😛 

* https://dom.spec.whatwg.org/#parent-element
* https://developer.mozilla.org/en-US/docs/Web/API/Node/parentElement

Reply:
ha - ya that makes sense. I would think they want it to be Element. Ok let's leave it for now.